cmd/agent/subcommands/flare/command.go

```diff
@@ -343,9 +357,30 @@ func makeFlare(flareComp flare.Component,
 		return err
 	}
 
+	enableStreamLog, settingErr := rcsetting.GetRuntimeSetting("enable_stream_logs")
+
+	if settingErr != nil {
+		fmt.Fprintln(color.Output, color.RedString(fmt.Sprintf("Error getting runtime setting: %s", settingErr)))
+	}
+
+	var duration time.Duration
+	var params *streamlogs.CliParams
+
 	if streamLogParams.Duration > 0 {
-		fmt.Fprintln(color.Output, color.GreenString((fmt.Sprintf("Asking the agent to stream logs for %s", streamLogParams.Duration))))
-		err := streamlogs.StreamLogs(lc, config, &streamLogParams)
+		duration = streamLogParams.Duration
+		params = &streamLogParams
+	}
+
+	if values, ok := enableStreamLog.([]interface{}); ok && len(values) > 1 {
+		if enable, ok := values[1].(bool); ok && enable {
+			duration = defaultRCDuration
+			params = &rcStreamLogParams
+		}
+	}
+
+	if duration > 0 {
+		fmt.Fprintln(color.Output, color.GreenString((fmt.Sprintf("Asking the agent to stream logs for %s", duration))))
+		err := streamlogs.StreamLogs(lc, config, params)
```

Comment:
How does this works: do we query the runtime agent to enable this ? Will the agent block for the duration ?

Reply:
I believe remote config would send a config remotely to the agent to collect a flare. While a flare is being asked to be generated, we would run additionally the streamlogs command while outputting it to a file (which is then attached to the flare). So unless the `datadog-agent stream-logs` command block the agent, i don't believe it would be blocked for the duration. 

As for how we query the runtime setting, once the setting is passed down by remote config, it should just be passed around as a component dependency. It should not block the agent unless my understanding of component and dependency is wrong. 
Let me know what you think.